### ISSUE TITLE
Using bigram instead of unigram decreases accuracy dramatically

### ISSUE THREAD

**Issue by lantianx2020** [facebookresearch/fastText#585]
Hi,
My training accuracy decreased dramatically when I used bigram/trigram instead of unigram. Does anyone know what's going on? 
Thanks!

**Comment by loretoparisi** [comment]
@lantianx2020 as far as I know, fasttext is using subwords information, so as soon as you are using NGram with an increasing N you will lose information.

**Comment by EdouardGrave** [comment]
Hi @lantianx2020,

Thank you for raising this issue.

One potential explanation is overfitting. What is the size of your training data? Could you please provide all the parameters that you used to train the models? Thank you for your help.

Best,
Edouard.

**Comment by matanster** [comment]
@EdouardGrave I just saw this after having been asked a similar question today. 
I think a possible reason may well be that one may have more unseen ngrams (in the supervised embedding underlying the training of the classification model), when increasing the _n_ of the ngrams from 1 (unigram) to 2 (bigram). 

Last I read the source, the implemented classification flow would not construct a vector for yet unseen ngrams. This may reduce the information available for the classifier during training and inference, if working a bigram model end-to-end is more likely to yield unknown ngrams than a unigram end-to-end model.

If I'm wrong here I'm happy to learn exactly where.
